common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.connector.kafka
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+import whisk.common.Logging
+import whisk.core.WhiskConfig
+import whisk.core.connector.MessageConsumer
+import whisk.core.connector.MessageProducer
+import whisk.core.connector.MessagingProvider
+import whisk.spi.Dependencies
+import whisk.spi.SingletonSpiFactory
+
+/**
+ * A Kafka based implementation of MessagingProvider
+ */
+class KafkaMessagingProvider() extends MessagingProvider {
+    def getConsumer(config: WhiskConfig, groupId: String, topic: String, maxPeek: Int, maxPollInterval: FiniteDuration)(implicit logging: Logging): MessageConsumer = {
+        new KafkaConsumerConnector(config.kafkaHost, groupId, topic, maxPeek, maxPollInterval = maxPollInterval)
+    }
+
+    def getProducer(config: WhiskConfig, ec:ExecutionContext)(implicit logging: Logging): MessageProducer = new KafkaProducerConnector(config.kafkaHost, ec)
+}
+
+object KafkaMessagingProvider extends SingletonSpiFactory[MessagingProvider] {
+    override def buildInstance(dependencies: Dependencies): MessagingProvider = new KafkaMessagingProvider
+}
```

Comment:
I ran into a similar issue with db, but in similar ways, it does not make sense for each entity store to be an spi (and use a different db), although it could be done that way. 

yes dependencies are optional currently and not used; I'm not sure of a great case for them but if there is an Spi that does NOT behave as a factory, that would be it. I would leave them for now.